.gitlab-ci.yml

```diff
@@ -22,22 +19,18 @@ variables:
 .build:                            &docker_build
   stage:                           build
   image:                           docker:stable
-  only:
-    refs:
-      - schedules
-      - web
-    kubernetes:                    active
+  rules:
+    # same as only: variables: $x
+    - if:                          '$DOCKERIMAGE == $CI_JOB_NAME'
+    # triggers rebuild when Dockerfile is changed
+    - changes:
+      - 'dockerfiles/${IMAGE_NAME}/Dockerfile'
   tags:
     - kubernetes-parity-build
   services:
     - docker:dind
   environment:
     name:                          parity-build
```

Comment:
afaiu the environment block can also go, since the jobs don't need to have access to the kubernetes api.